sqlite.zig

```diff
@@ -906,6 +906,17 @@ pub fn Iterator(comptime Type: type) type {
                     .Array => try self.readArray(FieldType, i),
                     .Pointer => try self.readPointer(FieldType, options.allocator, i),
                     .Optional => try self.readOptional(FieldType, options, i),
+                    .Enum => |TI| {
+                        const innervalue = try self.readField(FieldType.BaseType, i, options);
+
+                        if (comptime std.meta.trait.isZigString(FieldType.BaseType)) {
+                            return std.meta.stringToEnum(FieldType, innervalue) orelse @intToEnum(FieldType, 0);
```

Comment:
I wanted to do `orelse unreachable` here but the tests would not pass without handling a 0-value as the function was being passed an empty string. is this a bug in the tests or the appropriate behavior?

Reply:
note: tested with my own code where all column values are used and `unreachable` is not hit

Reply:
I've tried replacing with `orelse unreachable` and get a panic in [this test](https://github.com/vrischmann/zig-sqlite/blob/master/sqlite.zig#L1367-L1426). Is this what you have too ?

if yes, then the problem is a side effect of [this](https://github.com/vrischmann/zig-sqlite/pull/42/files#r694727123): `bindField` tries to bind column 6, which doesn't exist and the field `favorite_color` ends up being empty.

Reply:
that was it, thanks!